include/alpaka/dev/DevGenericSycl.hpp

```diff
@@ -119,14 +121,19 @@ namespace alpaka::experimental
 
         friend auto operator==(DevGenericSycl const& lhs, DevGenericSycl const& rhs) -> bool
         {
-            return (lhs.m_impl->m_device == rhs.m_impl->m_device) && (lhs.m_impl->m_context == rhs.m_impl->m_context);
+            return (lhs.m_impl == rhs.m_impl);
         }
 
         friend auto operator!=(DevGenericSycl const& lhs, DevGenericSycl const& rhs) -> bool
         {
             return !(lhs == rhs);
         }
 
+        [[nodiscard]] auto getNativeHandle() const noexcept -> std::pair<sycl::device, sycl::context>
```

Comment:
In this case we cannot guarantee that the method is `noexcept` (`std::pair`'s and the SYCL types' copy constructors are not `noexcept`).
```suggestion
        [[nodiscard]] auto getNativeHandle() const -> std::pair<sycl::device, sycl::context>
```